service/src/main/java/uk/gov/hmcts/reform/fpl/events/NoticeOfChangeEvent.java

```diff
@@ -4,13 +4,13 @@
 import lombok.Getter;
 import lombok.RequiredArgsConstructor;
 import uk.gov.hmcts.reform.fpl.model.CaseData;
-import uk.gov.hmcts.reform.fpl.model.RespondentSolicitor;
+import uk.gov.hmcts.reform.fpl.model.Respondent;
 
 @Getter
 @EqualsAndHashCode
 @RequiredArgsConstructor
 public class NoticeOfChangeEvent {
     private final CaseData caseData;
-    private final RespondentSolicitor oldRespondentSolicitor;
-    private final RespondentSolicitor newRespondentSolicitor;
+    private final Respondent oldRespondent;
```

Comment:
might be some early thinking ahead but instead of having them be of type respondent maybe there could be some interface that both the child and the respondent has that we can use to capture them. This way we can just have the interface designate the common menthods like `getSolicitor()` that are needed to build the notification and then when it comes to playing the story to plug everything in for the children it becomes much simpler

Reply:
As discussed, refactoring can be done in the Child notifications ticket